csrc/optimization/mark_alias.h

```diff
@@ -0,0 +1,20 @@
+// clang-format off
```

Comment:
nitpick. Leaning towards merge `mark_alias` with `alias_analysis`.

Reply:
Merge two files or merge the two classes? To the former, I general prefer creating one file per class to avoid unnecessary build dependency. To the latter, I want to separate analysis (which doesn't mutate a fusion) and passes (which mutate a fusion) and I do test the analysis separately.